odoo/addons/base/models/res_users.py

```diff
@@ -1377,12 +1375,11 @@ def _should_alert_new_device(self):
 # to the implied groups (transitively).
 #
 
-class GroupsImplied(models.Model):
-    _inherit = 'res.groups'
+class ResGroups(ResGroups):
 
-    implied_ids = fields.Many2many('res.groups', 'res_groups_implied_rel', 'gid', 'hid',
+    implied_ids: ResGroups = fields.Many2many(relation='res_groups_implied_rel', column1='gid', column2='hid',
         string='Inherits', help='Users of this group automatically inherit those groups')
-    trans_implied_ids = fields.Many2many('res.groups', string='Transitively inherits',
+    trans_implied_ids = fields.Many2many[ResGroups](string='Transitively inherits',
```

Comment:
If I remember correctly this form (`Many2many[Class]`) was better for type checkers. I think he former was rejected by type checkers because assigning a `Field` instance to an attribute of type `ResGroup` is not ok.

Reply:
Another form that is more elegant is `trans_implied_ids: Annotated[ResGroup, fields.Many2many(...)]`.  But that is not backward compatible.

Reply:
In case you have: `some_field_ids: ResGroup = fields.Many2many...` the typechecker will not accept it (same if you annotate): an error similar to expression has type "Many2Many[...]", variable has type "ResGroup".

`xxx = fields.Many2any[SomeModel]()` will typecheck correctly.